components/RecentlyPlayedDrawer.tsx

```diff
@@ -215,14 +379,14 @@ export default function RecentlyPlayedDrawer({}: RecentlyPlayedDrawerProps) {
     ),
   }));
 
-  const renderSong = (song: ProcessedSong, index: number) => {
+  const renderSong = (song: ProcessedSong, key: string) => {
     // Validate song data
     if (!song.title || !song.artist) {
       return null;
     }
 
     return (
-      <View key={`${song.title}-${song.artist}-${index}-${song.playedAt.getTime()}`} style={styles.songItem}>
+      <View key={`${key}-${song.title}-${song.artist}-${song.playedAt.getTime()}`} style={styles.songItem}>
```

Comment:
key can be simplified to just the passed-in key now - it's properly unique in all usages